### PR TITLE
Add product reduction for reduce op

### PR DESCRIPTION
**Pull request by zhouhanqing**
Finished #8129.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -2200,6 +2201,52 @@ def reduce_min(input, dim=None, keep_dim=False, name=None):
...
+        input (Variable): The input variable which is a Tensor or LoDTensor.
+        dim (int|None): The dimension along which the product is performed. If
+            :attr:`None`, multipy all elements of :attr:`input` and return a
+            Tensor variable with a single element, otherwise must be in the
+            range :math:`[-rank(input), rank(input))`. If :math:`dim < 0`,
+            the dimension to reduce is :math:`rank + dim`.
+        keep_dim (bool|False): Whether to reserve the reduced dimension in the
+            output Tensor. The result tensor will have one fewer dimension
+            than the :attr:`input` unless :attr:`keep_dim` is true.
+        name(str|None): A name for this layer(optional). If set None, the layer
```
Comment (pkuyym) [inline python/paddle/fluid/layers/nn.py]: Seems exceeds 80 columns.

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -2200,6 +2201,52 @@ def reduce_min(input, dim=None, keep_dim=False, name=None):
...
+        .. code-block:: python
+
+            # x is a Tensor variable with following elements:
+            #    [[0.2, 0.3, 0.5, 0.9]
+            #     [0.1, 0.2, 0.6, 0.7]]
+            # Each example is followed by the correspending output tensor.
+            fluid.layers.reduce_prod(x)  # [0.0002268]
+            fluid.layers.reduce_prod(x, dim=0)  # [0.02, 0.06, 0.3, 0.63]
+            fluid.layers.reduce_prod(x, dim=-1)  # [0.027, 0.0084]
+            fluid.layers.reduce_prod(x, dim=1, keep_dim=True)  # [[0.027], [0.0084]]
```
Comment (pkuyym) [inline python/paddle/fluid/layers/nn.py]: Same as above.

File: paddle/fluid/operators/reduce_op.cc [hunk 3]
```diff
@@ -173,6 +173,15 @@ class ReduceMinOpMaker : public ReduceOpMaker {
   }
 };
 
+class ReduceProdOpMaker : public ReduceOpMaker {
+ public:
+  ReduceProdOpMaker(OpProto *proto, OpAttrChecker *op_checker)
+      : ReduceOpMaker(proto, op_checker) {
+    SetComment("ReduceProd", "prod");
```
Comment (pkuyym) [inline paddle/fluid/operators/reduce_op.cc]: Please add more detailed comments.

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -2200,6 +2201,52 @@ def reduce_min(input, dim=None, keep_dim=False, name=None):
     return out
 
 
+def reduce_prod(input, dim=None, keep_dim=False, name=None):
+    """
+    Computes the product of tensor elements over the given dimension.
+
+    Args:
+        input (Variable): The input variable which is a Tensor or LoDTensor.
+        dim (int|None): The dimension along which the product is performed. If
```
Comment (pkuyym) [inline python/paddle/fluid/layers/nn.py]: dim --> axis ?

File: paddle/fluid/operators/reduce_op.cc (hunk 3 above)
Comment (pkuyym) [reply paddle/fluid/operators/reduce_op.cc]: prod --> production

**Comment by zhouhanqing** [comment]
Thank you for the review, and I'll modify these expressions.

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (zhouhanqing) [reply python/paddle/fluid/layers/nn.py]: I refer to the other reduce op's comments, like reduce_sum.

**Pull request merged**